spynnaker/pyNN/models/neuron/local_only/local_only_convolution.py

```diff
@@ -0,0 +1,269 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import numpy
+from collections import defaultdict, namedtuple
+from spinn_utilities.overrides import overrides
+from data_specification.enums.data_type import DataType
+from spinn_front_end_common.utilities.constants import (
+    BYTES_PER_SHORT, BYTES_PER_WORD)
+from spynnaker.pyNN.data import SpynnakerDataView
+from spynnaker.pyNN.exceptions import SynapticConfigurationException
+from spynnaker.pyNN.models.neural_projections.connectors import (
+    ConvolutionConnector)
+from spynnaker.pyNN.models.neuron.synapse_dynamics import (
+    AbstractSupportsSignedWeights)
+from spynnaker.pyNN.utilities.constants import SPIKE_PARTITION_ID
+from .abstract_local_only import AbstractLocalOnly
+
+Source = namedtuple("Source", ["projection", "vertex_slice", "key", "mask"])
+
+
+class LocalOnlyConvolution(AbstractLocalOnly, AbstractSupportsSignedWeights):
+    """ A convolution synapse dynamics that can process spikes with only DTCM
+    """
+
+    __slots__ = [
+        "__cached_2d_overlaps"
+    ]
+
+    def __init__(self):
+        # Store the overlaps between 2d vertices to avoid recalculation
+        self.__cached_2d_overlaps = dict()
+
+        # Store the merged keys for sources to avoid recalculation
+
+    @overrides(AbstractLocalOnly.merge)
+    def merge(self, synapse_dynamics):
+        if not isinstance(synapse_dynamics, LocalOnlyConvolution):
+            raise SynapticConfigurationException(
+                "All targets of this Population must have a synapse_type of"
+                " Convolution")
+        return synapse_dynamics
+
+    @overrides(AbstractLocalOnly.get_vertex_executable_suffix)
+    def get_vertex_executable_suffix(self):
+        return "_conv"
+
+    @property
+    @overrides(AbstractLocalOnly.changes_during_run)
+    def changes_during_run(self):
+        return False
+
+    @overrides(AbstractLocalOnly.get_parameters_usage_in_bytes)
+    def get_parameters_usage_in_bytes(
+            self, n_atoms, incoming_projections):
+        n_bytes = 0
+        for incoming in incoming_projections:
+            # pylint: disable=protected-access
+            s_info = incoming._synapse_information
+            if not isinstance(s_info.connector, ConvolutionConnector):
+                raise SynapticConfigurationException(
+                    "Only ConvolutionConnector can be used with a synapse type"
+                    " of Convolution")
+            # pylint: disable=protected-access
+            app_edge = incoming._projection_edge
+            n_incoming = len(
+                app_edge.pre_vertex.splitter.get_out_going_slices())
+            n_bytes += s_info.connector.local_only_n_bytes * n_incoming
+
+        return (6 * BYTES_PER_SHORT) + BYTES_PER_WORD + n_bytes
```

Comment:
A constant here for 6 (or 6 * BYTES_PER_SHORT) would be a good idea.

Reply:
Fixed